### ISSUE TITLE
CentOS 7: Installation of current rpy2 requires CFLAGS=-std=c99

### ISSUE THREAD

**Issue by dwt** [rpy2/rpy2#963]
When installing rpy2 on CentOS 7, with a newish (py3.8) python version, the following error occurs:

```
  rpy2/rinterface_lib/_bufferprotocol.c: In function 'memoryview_swapstrides':
  rpy2/rinterface_lib/_bufferprotocol.c:27:3: error: 'for' loop initial declarations are only allowed in C99 mode
     for (Py_ssize_t i=0; i < ndim; i++) {
     ^
  rpy2/rinterface_lib/_bufferprotocol.c:27:3: note: use option -std=c99 or -std=gnu99 to compile your code
  error: command '/usr/bin/gcc' failed with exit code 1
  ----------------------------------------
  ERROR: Failed building wheel for rpy2
```

Full reproduction here: https://gist.github.com/dwt/2a528a33319836ea9c48b8072da55aad

**Expected behavior**
Workaround for me was to set `export CFLAGS=-std=c99` before installing. Because that happens as a dependency of my application, that leads to me setting this for all dependencies, which has a big potential to create further bugs.

I would expect that rpy2 sets this value if needed by itself.

**Comment by lgautier** [comment]
Thanks. I never ran into it before. I am seeing that your are building the package for `aarch64`. This might what reveals the issue.

**Comment by lgautier** [comment]
Fixed with PR #964.
